diff --git a/node_modules/next/.bun-tag-45ddcc61e2178baf b/.bun-tag-45ddcc61e2178baf
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/next/.bun-tag-f3028f1c824ca116 b/.bun-tag-f3028f1c824ca116
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/server/lib/router-utils/proxy-request.js b/dist/server/lib/router-utils/proxy-request.js
index de8549da03855f8ac804a26a240184d443dec6fc..22feec259848cbba47e253a5657e02c9fe6eb639 100644
--- a/dist/server/lib/router-utils/proxy-request.js
+++ b/dist/server/lib/router-utils/proxy-request.js
@@ -1,4 +1,5 @@
 "use strict";
+const HttpProxy = require('next/dist/compiled/http-proxy');
 Object.defineProperty(exports, "__esModule", {
     value: true
 });
@@ -22,12 +23,12 @@ async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTime
     delete parsedUrl.query;
     parsedUrl.search = (0, _serverrouteutils.stringifyQuery)(req, query);
     const target = _url.default.format(parsedUrl);
-    const HttpProxy = require('next/dist/compiled/http-proxy');
     const proxy = new HttpProxy({
         target,
         changeOrigin: true,
         ignorePath: true,
         ws: true,
+        xfwd: true,
         // we limit proxy requests to 30s by default, in development
         // we don't time out WebSocket requests to allow proxying
         proxyTimeout: proxyTimeout === null ? undefined : proxyTimeout || 30000,
