diff --git a/dist/server/lib/router-utils/proxy-request.js b/dist/server/lib/router-utils/proxy-request.js
--- a/dist/server/lib/router-utils/proxy-request.js
+++ b/dist/server/lib/router-utils/proxy-request.js
@@ -1,4 +1,5 @@
 "use strict";
+require('next/dist/compiled/httpxy');
 Object.defineProperty(exports, "__esModule", {
     value: true
 });
@@ -27,12 +28,13 @@ async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTime
     // We just make sure this is a valid URL since httpxy would only parse it
     // lazily on the `web()`/`ws()` call.
     const target = new URL(_url.default.format(parsedUrl));
     const { ProxyServer } = require('next/dist/compiled/httpxy');
     const proxy = new ProxyServer({
         target,
         changeOrigin: true,
         ignorePath: true,
         ws: true,
+        xfwd: true,
         // we limit proxy requests to 30s by default, in development
         // we don't time out WebSocket requests to allow proxying
         proxyTimeout: proxyTimeout === null ? undefined : proxyTimeout || 30000,
@@ -48,6 +50,8 @@ async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTime
     // object will detect the disconnect, and we can abort the proxy's
     // connection.
     proxy.on('proxyReq', (proxyReq)=>{
+        proxyReq.removeHeader('accept-encoding');
+
         res.on('close', ()=>proxyReq.destroy());
     });
     proxy.on('proxyRes', (proxyRes)=>{
